frontend/js/tests/user/import-listens/ImportListens.test.tsx

```diff
@@ -0,0 +1,194 @@
+      expect(screen.getByRole("button", { name: /import listens/i })).toBeInTheDocument();
+    });
+  });
+
+  it("enables the import button after a file is uploaded", async () => {
+    renderWithProviders(<RouterProvider router={router} />, {}, { wrapper: ReactQueryWrapper }, false);
+    const importButton = screen.getByRole("button", { name: /import listens/i });
+    expect(importButton).toBeDisabled();
+
+    const file = new File(['{ "foo": "bar" }'], 'test.json', { type: 'application/json' });
+    const input = screen.getByLabelText(/choose a file/i);
+
+    fireEvent.change(input, { target: { files: [file] } });
+
+    await waitFor(() => {
+      const importButtonAfter = screen.getByRole("button", { name: /import listens/i });
+      expect(importButtonAfter).not.toBeDisabled();
+    });
+  });
+
+  it("renders imports with correct data", async () => {
+    renderWithProviders(<RouterProvider router={router} />, {}, { wrapper: ReactQueryWrapper }, false);
+    await waitFor(() => {
+      expect(screen.getByText("spotify.zip")).toBeInTheDocument();
+      expect(screen.getByText("August 11th, 2025")).toBeInTheDocument();
+      expect(screen.getByText("-")).toBeInTheDocument();
+    });
+  });
+
+it("calls import endpoint when import listens clicked", async () => {
+  let submitAPICalled = false;
+
+  server.use(
+    http.post("/settings/import/", (req) => {
+      submitAPICalled = true;
+      return HttpResponse.json({ user_has_email: true });
+    })
+  );
+
+  renderWithProviders(
+    <RouterProvider router={router} />,
+    {},
+    { wrapper: ReactQueryWrapper },
+    false
+  );
+
+  const input = screen.getByLabelText(/choose a file/i);
+
+  const file = new File(["spotify_test"], "spotify_test.zip", { type: "application/zip" });
+  fireEvent.change(input, { target: { files: [file] } });
+
+  const importButtonAfter = await screen.findByRole("button", { name: /import listens/i });
+  await waitFor(() => expect(importButtonAfter).not.toBeDisabled());
+
+  fireEvent.click(importButtonAfter);
+
+  await waitFor(() => expect(submitAPICalled).toBe(true));
+});
+
+
+  it("calls cancel endpoint when cancel clicked", async () => {
+
+    server.use(
+      http.post("/1/import-listens/cancel/1", (req) => {
+        cancelAPICalled = true;
+        return HttpResponse.json({ success: true, id: 1 });
+    }));
+    renderWithProviders(<RouterProvider router={router} />, {}, { wrapper: ReactQueryWrapper }, false);
+
+    await waitFor(() => {
+      expect(screen.getByText("spotify.zip")).toBeInTheDocument();
+    });
+
+    const cancelBtn = screen.getByRole("button", { name: "Cancel import" });
+    await user.click(cancelBtn);
+    await waitFor(() => expect(cancelAPICalled).toBe(true));
+  });
+
+  it("calls refresh endpoint when refresh clicked", async () => {
+    server.use(
+      http.get("/1/import-listens/1/", (req) => {
+        refreshAPICalled = true;
+        return HttpResponse.json(mockImports[0]);
```

Comment:
Here, to properly test the refreshing functionality, you would return an updated object, for example with an updated filename (won't happen in practice, but fine to test the mechanism):

```suggestion
        return HttpResponse.json(
          merge(mockImports[0], {
            metadata: { filename: "spotify-updated.zip" },
          })
        );
```

Test this before clicking the refresh button:
```
await waitFor(() => {
      expect(screen.getByText("spotify.zip")).toBeInTheDocument();
    });
    expect(screen.queryByText("spotify-updated.zip")).not.toBeInTheDocument();
```

And then after clicking the refresh button and checking refreshAPICalled:
```
    expect(screen.getByText("spotify-updated.zip")).toBeInTheDocument();
    expect(screen.queryByText("spotify.zip")).not.toBeInTheDocument();

```